qiskit/transpiler/passes/optimization/optimize_1q_commutation.py

```diff
@@ -220,9 +221,9 @@ def _step(self, dag):
                 dag,
                 (preceding_run or []) + run + (succeeding_run or []),
                 (
-                    (new_preceding_run or QuantumCircuit(1)).data
-                    + (new_run or QuantumCircuit(1)).data
-                    + (new_succeeding_run or QuantumCircuit(1)).data
+                    list(new_preceding_run.op_nodes())
+                    + list(new_run.op_nodes())
+                    + list(new_succeeding_run.op_nodes())
```

Comment:
In practice, `DAGCircuit.op_nodes` returns a `list` as things stand even though it should really return some lazy iterator.  Is this important enough for performance that we should avoid the extra list copies now at the cost of needing to update this when/if we fix `op_nodes`, or is it nowhere near a bottleneck?

Reply:
I'll drop the `list()` here, tbh I don't even remember why I did this. At one point I was struggling with types in this section with the new dag return and might have done this to try and workaround forgetting to update a caller.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/9188/commits/17bfe87857e33ec425067528f8429efc2ab5dfc1